src/pages/home/report/ReportActionsList.tsx

```diff
@@ -425,14 +426,49 @@ function ReportActionsList({
                     });
                     return;
                 }
+                const index = sortedVisibleReportActions.findIndex((item) => keyExtractor(item) === action?.reportActionID);
+                if (action?.actionName === CONST.REPORT.ACTIONS.TYPE.REPORT_PREVIEW) {
+                    if (index > 0) {
+                        setTimeout(() => {
+                            reportScrollManager.scrollToIndex(index);
+                        }, 100);
+                    } else {
+                        setIsFloatingMessageCounterVisible(false);
+                        reportScrollManager.scrollToBottom();
+                    }
+                    if (action?.reportActionID) {
+                        setActionIdHighlightedTemporary(action.reportActionID);
+                    }
+                } else {
+                    setIsFloatingMessageCounterVisible(false);
+                    reportScrollManager.scrollToBottom();
+                }
 
-                setIsFloatingMessageCounterVisible(false);
-                reportScrollManager.scrollToBottom();
                 setIsScrollToBottomEnabled(true);
             });
         },
-        [report.reportID, reportScrollManager, setIsFloatingMessageCounterVisible],
+        [report.reportID, reportScrollManager, setIsFloatingMessageCounterVisible, sortedVisibleReportActions, reportScrollManager],
     );
+
+    // Clear highlight report action after scroll and highlight
+    useEffect(() => {
+        if (actionIdHighlightedTemporary === '') {
+            return;
+        }
+        // Time highlight is the same as SearchPage
+        const duration =
+            CONST.ANIMATED_HIGHLIGHT_ENTRY_DELAY +
+            CONST.ANIMATED_HIGHLIGHT_ENTRY_DURATION +
+            CONST.ANIMATED_HIGHLIGHT_START_DELAY +
+            CONST.ANIMATED_HIGHLIGHT_START_DURATION +
+            CONST.ANIMATED_HIGHLIGHT_END_DELAY +
+            CONST.ANIMATED_HIGHLIGHT_END_DURATION;
```

Comment:
Hmm I don't love having this duplicated from SearchList.tsx... could we extract it somewhere so that the list only exists once?